### PR TITLE
bg/meson-fixes

### PR DESCRIPTION
**Pull request by bengal**
Some other meson-related fixes.

### CODE REVIEW

**Comment by thom311** [comment]
Generally lgtm.

However, follow-up issues:

---

```
$ meson build -D introspection=false 

meson.build:717:4: ERROR:  Problem encountered: Vala bindings require GObject introspection
```
This should automatically disable vapi.

---
```
meson build -D introspection=false -D vapi=false -D docs=true
```
should not work, because documentation requires introspection.

**Comment by bengal** [comment]
Fixed some issues and repushed.

File: tools/meson-post-install.sh [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+localstatedir=$5
+
+[ -n "$DESTDIR" ] && DESTDIR=${DESTDIR%%/}/
+
+mv "${DESTDIR}${datadir}/bash-completion/completions/nmcli-completion" \
+   "${DESTDIR}${datadir}/bash-completion/completions/nmcli"
+
+for alias in nmtui-connect nmtui-edit nmtui-hostname; do
+    ln -sf nmtui "${DESTDIR}${bindir}/$alias"
+done
```
Comment (thom311) [inline tools/meson-post-install.sh]: if tui isn't enabled, these symlinks shouldn't be installed.

**Comment by thom311** [comment]
With
```
  $ meson build -D introspection=false -D vapi=true
```
we get
```
meson.build:717: WARNING: Vala bindings are disabled because they require GObject introspection
```
but it seems to me a hard failure would be in order.

File: tools/meson-post-install.sh (hunk 1 above)
Comment (bengal) [reply tools/meson-post-install.sh]: Fixed.

**Comment by bengal** [comment]
> but it seems to me a hard failure would be in order.

Yeah, I changed the option to a auto|true|false choice.

**Comment by thom311** [comment]
lgtm

**Pull request merged**